src/expandProperty.js

```diff
@@ -1,41 +1,57 @@
-const WHITESPACE_NO_CALC = /\s+(?=[^)]*?(?:\(|$))/g
-const LENGTH_UNIT = /(calc\(|em|ex|ch|rem|vw|vh|vmin|vmax|cm|mm|q|in|pt|pc|px|dpi|dpcm|dppx|%|auto)$/i
+const WHITESPACE_NO_PARENS = /\s+(?=[^)]*?(?:\(|$))/g
+const LENGTH_UNIT = /(em|ex|ch|rem|vw|vh|vmin|vmax|cm|mm|q|in|pt|pc|px|dpi|dpcm|dppx|%|auto)$/i
+const EDGE_CASES = /^(calc\(|var--)/i
+const CUSTOM_PROPERTIES = /var\((--[\w-]+)\)/g
 
 const BORDER_STYLE = /^(dashed|dotted|double|groove|hidden|inset|none|outset|ridge|solid)$/i
 const BORDER_WIDTH = /^(thick|medium|think)$/i
 
-function parseBorder(value, resolve) {
-  const values = value.split(WHITESPACE_NO_CALC)
+// Replace `var(--foo)` with `var--foo` to remove the nested parens issues
+function removeCustomProperties(value) {
+  if (value.indexOf('var(') === -1) return value
```

Comment:
I’m not super sure whether this will actually speed things up at all. I’d assume the JavaScript engine would do an initial check on the string to optimise performance in case of null hit; if it does, we’d ping the string twice instead of once only, which might in turn be worse for performance. Genuinely not too sure, but either way we’re really nitpicking here.